app/src/main/java/de/westnordost/streetcomplete/quests/air_pump/AddAirCompressor.kt

```diff
@@ -0,0 +1,49 @@
+package de.westnordost.streetcomplete.quests.air_pump
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CAR
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+import de.westnordost.streetcomplete.util.ktx.toYesNo
+import de.westnordost.streetcomplete.osm.updateWithCheckDate
+
+class AddAirCompressor : OsmFilterQuestType<Boolean>() {
+
+    override val elementFilter = """
+       nodes, ways with
+       amenity = fuel
+       and (
+           !compressed_air and !service:bicycle:pump
+           or compressed_air older today -6 years
+       )
+       and access !~ private|no
+    """
+    /* if service:bicycle:pump is undefined, nothing has been said about its existence;
+     * see https://wiki.openstreetmap.org/wiki/Tag:shop=bicycle#Additional_keys */
+
+    override val changesetComment = "Add whether air compressor is available"
+    override val wikiLink = "Key:compressed_air"
+    override val icon = R.drawable.ic_quest_air_pump
+    override val questTypeAchievements = listOf(CAR, BICYCLIST)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_air_pump_compressor_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("""
+            nodes, ways with
+            compressed_air
+            or service:bicycle:pump
+            or amenity ~ compressed_air|fuel|bicycle_repair_station
+        """)
```

Comment:
Why highlight fuel stations regardless of if they have compressed air or not? This is not consistent. On the one side, you highlight **any** map feature with `compressed_air` (even set to `no`), on the other hand you highlight map features that might or might not have compressed air. Additionally, you highlight any manual bicycle pumps but the question is not about manual air pumps. This too is misleading.

```suggestion
        getMapData().filter("nodes, ways with compressed_air = yes or amenity = compressed_air")
```
... will highlight any feature with compressed air or separately mapped compressed air.

Reply:
I guess you're right on showing manual bicycle pumps, I probably see things too much from bicyclist point of view, so copy/pasting that code seemed logical. But now that you mention it, even if fuel station has manual bicycle pump, it would probably be on separately marked bicycle_repair_station which is covered by the other quest. So I've corrected that part in c89e209dc136330031ef2c4ef20705bd2e868cee.

Reply:
ok